### PR TITLE
Partition pruning documentation

### PR DESCRIPTION
**Pull request by Fly-Style**
WIP

### CODE REVIEW

File: docs/modules/sql/pages/partition-pruning.adoc [hunk 1]
```diff
@@ -0,0 +1,39 @@
+= Partition pruning of SQL queries
+:description: Use partition pruning optimization technique to reduce latency for quick IMap SQL queries.
+:page-beta: false
+
+Partition pruning is an IMap-specific query optimization technique that automatically prunes members which don't contain
+the required data on their partitions, and optimize count of Jet scan processors on required members. It works
+
+* by default, for primitive `__key` types (default attribute strategy);
+* for all key types if user configure attribute partitioning strategy for `MapConfig`.
```
[the remark this answers is not in the document]
Comment (Fly-Style) [reply docs/modules/sql/pages/partition-pruning.adoc:9]: We call this entity `AttributePartitioningStrategy`, so, I'll use the full name.

File: docs/modules/sql/pages/partition-pruning.adoc [hunk 2]
```diff
@@ -0,0 +1,63 @@
+= Partition pruning of SQL queries
+:description: The partition pruning optimization technique reduces latency for faster IMap SQL queries.
+:page-beta: false
+
+Partition pruning is an IMap-specific query optimization technique that automatically prunes members that don't contain
+the required data on their partitions, and optimizes the count of Jet scan processors on required members. It works:
+
+* by default, for primitive `__key` types (default attribute strategy).
+* for all key types if you configure the `AttributePartitioningStrategy` in `MapConfig`.
+
```
Comment (rebekah-lawrence) [inline docs/modules/sql/pages/partition-pruning.adoc:10]: Suggest amalgamating the notes, as follows:

```
[NOTE] 
====
. You can only use partition pruning if the partition strategy attribute is enabled.
. The query.....
====
```

Comment (rebekah-lawrence) [reply docs/modules/sql/pages/partition-pruning.adoc:10]: Is there/will there be documentation for the AttributePartitionStrategy related config? If so, a link would be good

File: docs/modules/sql/pages/partition-pruning.adoc [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+NOTE: The partition pruning may be used *only* when querying an IMap mapping if the partitioning strategy attribute is enabled.
+
+NOTE: The query coordinator member is always involved in the query execution, even if it doesn't contain any
+required data.
+
+== Example
+
```
Comment (rebekah-lawrence) [inline docs/modules/sql/pages/partition-pruning.adoc:17]: can we have some intro text, something like "The following example shows the configuration updates for......." 

and include information on the filename/location requirements of the config file as well?

Comment (rebekah-lawrence) [reply docs/modules/sql/pages/partition-pruning.adoc:17]: As there are three examples, we need to add context for each

**Review by rebekah-lawrence — CHANGES_REQUESTED** [review]

File: docs/modules/sql/pages/partition-pruning.adoc (hunk 3 above)
Comment (Fly-Style) [reply docs/modules/sql/pages/partition-pruning.adoc:17]: > and include information on the filename/location requirements of the config file as well?

We don't have such for partition pruning. :)

File: docs/modules/sql/pages/partition-pruning.adoc (hunk 2 above)
Comment (Fly-Style) [reply docs/modules/sql/pages/partition-pruning.adoc:10]: I made a quick search, and doc for `AttributePartitionStrategy` doesn't exist yet.

**Review by rebekah-lawrence — APPROVED** [review]
LGTM
In the future, a link will be needed to the  AttrbutePartitionStrategy information, when available.

**Review by Serdaro — APPROVED** [review]
LGTM

**Pull request merged**